### PR TITLE
OHRM5X-990 - Duplicate project activity validation

### PR DESCRIPTION
**Pull request by SandeepaOHRM**
This PR includes:
Back-end API for duplicate project activity validation
Integrated testing for API
Unit test for DAO

### CODE REVIEW

File: symfony/plugins/orangehrmTimePlugin/config/routes.yaml [hunk 1]
```diff
@@ -1,116 +1,124 @@
 apiv2_time_project:
   path: /api/v2/time/projects/{id}
   controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
-  methods: [ GET, PUT ]
+  methods: [GET, PUT]
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/config/routes.yaml:4]: format this file using IDE

File: symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php [hunk 2]
```diff
@@ -970,4 +970,26 @@ public function saveAndUpdateTimesheetItems(array $timesheetItems): void
...
+    ): bool {
+        $qb = $this->createQueryBuilder(TimesheetItem::class, 'timesheetItem');
+        $qb->andWhere('timesheetItem.timesheet = :timesheetId');
+        $qb->setParameter('timesheetId', $timesheetId);
+        $qb->andWhere('timesheetItem.project = :projectId');
+        $qb->setParameter('projectId', $projectId);
+        $qb->andwhere('timesheetItem.projectActivity = :activityId');
+        $qb->setParameter('activityId', $activityId);
+
+        return $this->getPaginator($qb)->count() === 0;
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php:993]:
```suggestion
        return $this->getPaginator($qb)->count() > 0;
```
I feel according to the method name `isDuplicateTimesheetItem` logic should change like this. What do you think?

File: symfony/plugins/orangehrmTimePlugin/test/fixtures/TimesheetProjectActivityAPITest.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+Employee:
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/test/fixtures/TimesheetProjectActivityAPITest.yml:1]: better to add config values. seems ur tests depend on other fixtures

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: installer/dbscript/dbscript-2.sql [hunk 4]
```diff
@@ -4250,6 +4250,7 @@ VALUES ('apiv2_time_time_sheet_config', 'API-v2 Time - Time Sheet Start Day Conf
        ('apiv2_time_my_timesheets_entries', 'API-v2 Time - Timesheet Entries', 1, 0, 1, 0),
        ('apiv2_time_my_timesheet', 'API-v2 Time - My Timesheet', 1, 1, 1, 0),
        ('apiv2_time_timesheet_action_log', 'API-v2 Time - Timesheet action log', 1, 0, 0, 0),
+       ('apiv2_time_project_activity_unique_validation', 'API-v2 Time - project Activity Unique Validation', 1, 0, 0, 0),
```
Comment (RajithaKumara) [inline installer/dbscript/dbscript-2.sql:4253]:
```suggestion
       ('apiv2_time_project_activity_unique_validation', 'API-v2 Time - Project Activity Unique Validation', 1, 0, 0, 0),
```

File: symfony/plugins/orangehrmTimePlugin/config/routes.yaml (hunk 1 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/config/routes.yaml:4]: done

File: symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php (hunk 2 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php:993]: greater than 0 means there can be 1 or more neh?? then we are assuming that there might be already a duplicate record

File: symfony/plugins/orangehrmTimePlugin/test/fixtures/TimesheetProjectActivityAPITest.yml (hunk 3 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/test/fixtures/TimesheetProjectActivityAPITest.yml:1]: done

File: installer/dbscript/dbscript-2.sql (hunk 4 above)
Comment (SandeepaOHRM) [reply installer/dbscript/dbscript-2.sql:4253]: done

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**